js/src/main/scala/java/util/TimeZone.scala

```diff
@@ -52,11 +55,16 @@ object TimeZone {
   def getDefault: TimeZone = default
   def setDefault(timeZone: TimeZone): Unit = default = timeZone
 
-  def getTimeZone(timeZone: String): TimeZone = ???
-  def getTimeZone(zoneId: ZoneId): TimeZone   = ???
+  def getTimeZone(timeZone: String): TimeZone = getTimeZone(ZoneId.of(timeZone))
+  def getTimeZone(zoneId: ZoneId): TimeZone   = {
+    val rules = ZoneRulesProvider.getRules(zoneId.getId, forCaching = true)
+    val offsetInMillis = rules.getOffset(Instant.now).getTotalSeconds * 1000
+    new SimpleTimeZone(offsetInMillis, zoneId.getId)
+  }
 
-  def getAvailableIDs: Array[String] = ???
-  def getAvailableIDs(offsetMillis: Int): Array[String] = ???
+  def getAvailableIDs: Array[String] = ZoneRulesProvider.getAvailableZoneIds.asScala.toArray
+  def getAvailableIDs(offsetMillis: Int): Array[String] =
+    getAvailableIDs.view.map(getTimeZone).filter(_.getRawOffset == offsetMillis).map(_.getID).toArray
```

Comment:
The purpose of the `view` here is to fuse the `map(...).filter(...).map(...)` in one single traverse instead of three... but yeah, I could operate in the array straight.